src/system/file_path.hh

```diff
@@ -19,6 +19,7 @@
 #define FILE_NAME_HH_
 
 #include <string>
+#include <boost/filesystem.hpp>
```

Comment:
We can move implementation of operators to cc file, but we can't remove include, because FilePath class contains boost::filesystem::path class members.